SLIDEseq dir schema

mccalluc opened the pull request:
Fix #454

mccalluc commented on src/ingest_validation_tools/directory-schemas/slideseq.yaml:
```diff
@@ -0,0 +1,44 @@
+-
+  pattern: '[^/]+/alignment/Puck_\w+\.digital_expression\.txt\.gz'
+  description: 'bead x gene expression matrix, ~25M'
+-
+  pattern: '[^/]+/alignment/Puck_\w+\.exonic\+intronic\.pdf'
+  description: 'plots of sequencing data, alignment and barcode matching, ~3M'
```
```suggestion
-
  pattern: '[^/]+/alignment/Puck_\w+\.exonic\+intronic\.pdf'
  description: 'plots of sequencing data, alignment and barcode matching, ~3M!!!'
```

pdblood commented:
@mccalluc I discussed with @ropelews and he would like Broad to upload an example dataset as a reference that he can then work with them on to finalize the directory structure. FYI, he is currently working with Florida to confirm/update the directory structures for data types they are submitting.

ropelews reviewed (commented):
Suggested replacement -- note definition clarified and addition of fastq directory.
A dataset is a "Puck"
I can write some draft suggested file/directory name guidelines to pass on to the new tmc's (like using all lower case, avoiding spaces, commas, asterisks in filenames.)
```
-
#
# subdirectory alignment
#
pattern: 'experiment-date_library-ID[^/]+/alignment/Puck_[^/]+\.bam'
description: 'aligned sequencing data from Slide-seq experiments against reference HG38'
-
pattern: 'experiment-date_library-ID[^/]+/alignment/Puck_[^/]+_mapping_rate\.txt'
description: 'mapping rate summary (~ 10 number of mapping statistics per puck)'
-
pattern: 'experiment-date_library-ID[^/]+/alignment/Puck_[^/]+_alignment_quality\.pdf'
description: 'mapping quality plots (has unique and multiple alignment ratio, alignment scores alignment mismatch)'
-
pattern: 'experiment-date_library-ID[^/]+/alignment/Puck_[^/]+.digital_expression\.txt\.gz'
description 'bead x gene expression matrix (csv file)'
-
pattern: 'experiment-date_library-ID[^/]+/alignment/Puck_[^/]+\.exonic+intronic\.pdf'
description: 'post alignment plots (qa/qc), plots of sequencing data, alignment and barcode matching'
-
#
# subdirectory barcode_matching
#
pattern: 'experiment-date_library-ID[^/]+/barcode_matching/BeadBarcodes\.txt'
description: 'barcodes of all sequenced beads (Many of these barcodes will not be in the matched_bead_barcodes files)'
-
pattern: 'experiment-date_library-ID[^/]+/barcode_matching/BeadLocations\.txt'
description: 'spatial coordinates of all sequenced beads (1 to 1 correspondence with BeadBarcodes)'
-
pattern: 'experiment-date_library-ID[^/]+/barcode_matching/Puck_[^/]+_unique_matched_illumina_barcodes\.txt'
description: 'matched Illumina barcodes (used by illumina sequencer)'
-
pattern: 'experiment-date_library-ID[^/]+/barcode_matching/Puck_[^/]+_matched_bead_barcodes\.txt'
description: 'matched bead barcodes (these are the barcodes that matched bead on the puck)'
-
pattern: 'experiment-date_library-ID[^/]+/barcode_matching/Puck_[^/]+_matched_bead_locations\.txt'
description: 'matched bead coordinates (these are the location of the barcodes that matched bead on the puck)'
-
#
# subdirectory fastq
#
pattern: 'experiment-date_library-ID[^/]+/fastq/Puck_[^/]+read1.fastq.gz'
description: 'each puck will have 2 fastq files, this file contains the first set of paired reads'
-
pattern: 'experiment-date_library-ID[^/]+/fastq/Puck_200906_14.read2.fastq.gz'
description: 'each puck will have 2 fastq files, this file contains the second set of paired reads'
-
```

mccalluc commented:
@ropelews : Thanks: I've integrated most of your suggestions. Notes:
- When you have a multiline block of code in a comment here, surround it with <code>```</code> before and after.
- I don't think you want the directory name to be literally `experiment-date_library-ID`, so I've removed that. We could have some kind of restriction on the directory names, but my sense it would be more noise, without improving the quality of the submission.
- Longer descriptions are great!
- I've taken your loosening of the `Puck_...` string... now it can be anything except a `/`
- I've replaced backslashes in `\+` and `\.`: both plus and period are metacharacters, so they need to be quoted.
